### ISSUE TITLE
How did you pre-process the dataset

### ISSUE THREAD

**Issue by carlosh93** [vincenzosantopietro/action-recognition-efficientnet#1]
Hi, thank you for this repository. How did you pre-process the dataset? Would you please share the method or upload the preprocessed dataset to another storage service (like google drive, for instance)? I am having trouble with LFS. I think it would be more useful if you share the preprocessing steps. Thank you!

**Comment by vincenzosantopietro** [comment]
Hi Carlos,
thank you for spending your time visiting and reviewing this repo.

1 - DATASET
I've used the UCF101 dataset that I found at this Kaggle page (but i noticed it doesn't exist anymore): https://www.kaggle.com/ash81197/vid-classification-ucf101/version/1

2 - PREPROCESSING
As you can see from this piece of code in data_loader.py
```python 
self.train_ds = tf.keras.preprocessing.image_dataset_from_directory(
            self.training_data_path,
            validation_split=0.2,
            subset="training",
            seed=345,
            image_size=(self.image_shape[0], self.image_shape[1]),
            batch_size=self.batch_size,
            label_mode='categorical',
            class_names=self.classes
        )

        self.val_ds = tf.keras.preprocessing.image_dataset_from_directory(
            self.training_data_path,
            validation_split=0.2,
            subset="validation",
            seed=345,
            image_size=(self.image_shape[0], self.image_shape[1]),
            batch_size=self.batch_size,
            label_mode='categorical',
            class_names=self.classes
        )

        self.test_ds = tf.keras.preprocessing.image_dataset_from_directory(
            self.test_data_path,
            seed=123,
            image_size=(self.image_shape[0], self.image_shape[1]),
            batch_size=self.batch_size,
            label_mode='categorical',
            class_names=self.classes
        )
 
        normalization_layer = tf.keras.layers.experimental.preprocessing.Rescaling(1. / 255)
        self.train_ds = self.train_ds.map(lambda x, y: (normalization_layer(x), y))
        self.val_ds = self.train_ds.map(lambda x, y: (normalization_layer(x), y))
        self.test_ds = self.test_ds.map(lambda x, y: (normalization_layer(x), y))
```
preprocessing is just a resizing of the images to whatever dimension you like + a simple normalization of pixel values.

I've  uploaded the dataset to this GDrive Link: https://drive.google.com/file/d/1cLC1ie8uVr51RBtQaDnEmI8kfO0oZy9Y/view?usp=sharing
Make sure to save this dataset as i'm finishing space on Drive and i'll remove it soon.
